test/albums.js

```diff
@@ -120,3 +140,133 @@ describe('/albums GET', () => {
+        expect(err.response.body.message).to.equal('Missing token.');
+        expect(err.response.body.internal_code).to.equal('invalid_token');
+        done();
+      });
+  });
+
+  it('should fail because token is invalid', done => {
+    chai
+      .request(server)
+      .post('/albums/1')
+      .set(token.headerName, token.encode({ email: 'all your base belong to us' }))
+      .catch(err => {
+        err.should.have.status(401);
+        err.response.body.should.have.property('message');
+        err.response.body.should.have.property('internal_code');
+        expect(err.response.body.message).to.equal('Invalid token.');
+        expect(err.response.body.internal_code).to.equal('invalid_token');
+        done();
+      });
+  });
+
+  before(() => {
+    nock.cleanAll();
+    noAlbum();
+  });
+
+  it('should fail because external service is unavailable', done => {
+    factory.create('user').then(user => {
+      chai
+        .request(server)
+        .post('/albums/101')
+        .set(token.headerName, token.encode({ email: user.email }))
+        .catch(err => {
+          err.should.have.status(404);
+          err.response.body.should.have.property('message');
+          err.response.body.should.have.property('internal_code');
+          expect(err.response.body.internal_code).to.equal('fetch_error');
+          done();
+        });
+    });
+  });
+
+  after(() => {
+    nock.cleanAll();
+  });
+
+  before(() => {
+    nock.cleanAll();
+    noAlbum(false);
+  });
+
+  it('should fail because external album does not exist', done => {
+    factory.create('user').then(user => {
+      chai
+        .request(server)
+        .post('/albums/101')
+        .set(token.headerName, token.encode({ email: user.email }))
+        .catch(err => {
+          err.should.have.status(404);
+          err.response.body.should.have.property('message');
+          err.response.body.should.have.property('internal_code');
+          expect(err.response.body.internal_code).to.equal('fetch_error');
+          done();
+        });
+    });
+  });
+
+  before(() => {
+    nock.cleanAll();
+    oneAlbum();
+  });
+
+  it('should fail because user purchased the same album twice', done => {
+    factory.create('user').then(user => {
+      chai
+        .request(server)
+        .post('/albums/1')
+        .set(token.headerName, token.encode({ email: user.email }))
+        .then(() => {
+          chai
+            .request(server)
+            .post('/albums/1')
+            .set(token.headerName, token.encode({ email: user.email }))
+            .catch(err => {
+              err.should.have.status(400);
+              err.response.body.should.have.property('message');
+              err.response.body.should.have.property('internal_code');
+              expect(err.response.body.internal_code).to.equal('saving_error');
+              expect(err.response.body.message).to.equal('Validation error');
```

Comment:
Add some description in this message (duplicated album, album already purchased, etc)